### PR TITLE
convert pointer to ptr for sig-apps integration tests

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?

/kind deprecation
/kind cleanup
#### What this PR does / why we need it:

Convert some of the integration tests to use ptr rather than pointer.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by kannon92** [comment]
@atiratree are you able to take a look at this PR?

File: test/integration/job/job_test.go [hunk 1]
```diff
@@ -1694,45 +1694,45 @@ func TestJobPodReplacementPolicy(t *testing.T) {
...
 			jobSpec: &batchv1.JobSpec{
-				Parallelism:          pointer.Int32Ptr(podCount),
-				Completions:          pointer.Int32Ptr(podCount),
+				Parallelism:          ptr.To(podCount),
+				Completions:          ptr.To(podCount),
 				CompletionMode:       &indexedCompletion,
 				PodReplacementPolicy: podReplacementPolicy(batchv1.TerminatingOrFailed),
 			},
-			wantTerminating: pointer.Int32(podCount),
+			wantTerminating: ptr.To[int32](podCount),
```
Comment (atiratree) [inline test/integration/job/job_test.go:1708]: Inconsistent usage here and in other places. We do not need to specify the type when the type of the variable is known.

**Comment by atiratree** [comment]
apart from the nit, LGTM

File: test/integration/job/job_test.go (hunk 1 above)
Comment (kannon92) [reply test/integration/job/job_test.go:1708]: Thanks for catching that.

Comment (kannon92) [reply test/integration/job/job_test.go]: pushed a fix for that.

File: test/integration/job/job_test.go [hunk 2]
```diff
@@ -343,8 +343,8 @@ func TestJobPodFailurePolicyWithFailedPodDeletedDuringControllerRestart(t *testi
 					},
 				},
 			},
-			Parallelism: pointer.Int32(int32(count)),
-			Completions: pointer.Int32(int32(count)),
+			Parallelism: ptr.To[int32](int32(count)),
+			Completions: ptr.To[int32](int32(count)),
```
Comment (atiratree) [inline test/integration/job/job_test.go:347]: found a couple more: casting should be enough in these two

File: test/integration/job/job_test.go [hunk 3]
```diff
@@ -1929,8 +1929,8 @@ func TestElasticIndexedJob(t *testing.T) {
 			mode := batchv1.IndexedCompletion
 			jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
 				Spec: batchv1.JobSpec{
-					Parallelism:    pointer.Int32Ptr(initialCompletions),
-					Completions:    pointer.Int32Ptr(initialCompletions),
+					Parallelism:    ptr.To[int32](initialCompletions),
+					Completions:    ptr.To[int32](initialCompletions),
```
Comment (atiratree) [inline test/integration/job/job_test.go:1933]: has same type here

File: test/integration/job/job_test.go [hunk 4]
```diff
@@ -2022,8 +2022,8 @@ func BenchmarkLargeIndexedJob(b *testing.B) {
 						Name: fmt.Sprintf("npods-%d-%d", nPods, n),
 					},
 					Spec: batchv1.JobSpec{
-						Parallelism:    pointer.Int32(nPods),
-						Completions:    pointer.Int32(nPods),
+						Parallelism:    ptr.To[int32](nPods),
+						Completions:    ptr.To[int32](nPods),
```
Comment (atiratree) [inline test/integration/job/job_test.go:2026]: here

File: test/integration/job/job_test.go [hunk 5]
```diff
@@ -2393,9 +2393,9 @@ func TestSuspendJob(t *testing.T) {
 			parallelism := int32(2)
 			job, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
 				Spec: batchv1.JobSpec{
-					Parallelism: pointer.Int32(parallelism),
-					Completions: pointer.Int32(4),
-					Suspend:     pointer.BoolPtr(tc.create.flag),
+					Parallelism: ptr.To[int32](parallelism),
```
Comment (atiratree) [inline test/integration/job/job_test.go:2396]: here

File: test/integration/job/job_test.go (hunk 3 above)
Comment (kannon92) [reply test/integration/job/job_test.go]: updated.

File: test/integration/job/job_test.go (hunk 4 above)
Comment (kannon92) [reply test/integration/job/job_test.go]: updated.

**Review by atiratree — APPROVED** [review]

**Review by alculquicondor — COMMENTED** [review]
/approve
/lgtm

**Comment by kannon92** [comment]
I just resolved the conflicts and pushed these changes.  I notice some flakiness with the rebase label getting removed.

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**